streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -213,11 +211,11 @@ public boolean isValidTransition(final State newState) {
     private final Object stateLock = new Object();
     protected volatile State state = State.CREATED;
 
-    private boolean waitOnState(final State targetState, final long waitMs) {
+    private boolean waitOnNoRunningState(final long waitMs) {
```

Comment:
I'd prefer keep it as general so that the same function can be reused in the future potentially rather than explicitly restrict is usage scope.